apis/Google.Cloud.Spanner.V1/Google.Cloud.Spanner.V1/SpannerDate.cs

```diff
@@ -0,0 +1,260 @@
+        /// </summary>
+        /// <remarks>
+        /// <para>
+        /// <paramref name="text"/> must be a representation of a date value in yyyy-MM-dd format which can be represented by <see cref="SpannerDate"/>.
+        /// </para>
+        /// </remarks>
+        /// <param name="text">The text to parse. Must not be null. Must be in yyyy-MM-dd format only.</param>
+        /// <returns>The parsed value.</returns>
+        /// <exception cref="FormatException">The value could not be parsed as a <see cref="SpannerDate"/>.</exception>
+        public static SpannerDate Parse(string text)
+        {
+            string message = TryParseImpl(text, out var value);
+            return message == null ? value : throw new FormatException(message);
+        }
+
+        /// <summary>
+        /// Attempts to parse a textual representation of date in yyyy-MM-dd format as <see cref="SpannerDate"/>.
+        /// </summary>
+        /// <remarks>
+        /// See <see cref="Parse(string)"/> for format details. This method will return <c>true</c> if and only if
+        /// <see cref="Parse(string)"/> would return without an exception.
+        /// </remarks>
+        /// <param name="text">The text to parse. Must not be null. Must be in yyyy-MM-dd format only.</param>
+        /// <param name="value">The parsed value, or 0 on failure.</param>
+        /// <returns><c>true</c> if <paramref name="text"/> was parsed successfully; <c>false</c> otherwise.</returns>
+        public static bool TryParse(string text, out SpannerDate value)
+        {
+            string message = TryParseImpl(text, out value);
+            return message == null;
+        }
+
+        /// <summary>
+        /// Returns the error message for a FormatException if parsing fails, or null for success.
+        /// </summary>
+        private static string TryParseImpl(string text, out SpannerDate value)
+        {
+            GaxPreconditions.CheckNotNull(text, nameof(text));
+
+            if (DateTime.TryParseExact(text, "yyyy-MM-dd", CultureInfo.InvariantCulture, DateTimeStyles.None, out DateTime dateTime))
+            {
+                value = new SpannerDate(dateTime);
+                return null;
+            }
+            else
+            {
+                value = default;
+                return "Text representation must be a valid date.";
+            }
+        }
+
+        /// <summary>
+        /// Returns the ISO 8601 representation (yyyy-MM-dd) of the date represented by this instance.
+        /// </summary>
+        /// <returns>The ISO 8601 string representation of the date represented by this instance.</returns>
+        public override string ToString() => ToDateTime().ToString("yyyy-MM-dd", CultureInfo.InvariantCulture);
+
+        /// <summary>
+        /// Returns a SpannerDate instance that is set to the date part of the specified DateTime.
+        /// </summary>
+        /// <param name="dateTime">The date time.</param>
```

Comment:
I'd change this to `The DateTime value.` (as you've got it below) - because "date time` itself isn't a meaningful English phrase.